### PR TITLE
When enforcing version_limit, order the version records by created_at

### PR DESCRIPTION
**Pull request by danbernier**
The details of the issue are in #940.

I split the commits out into visible, obvious parts. If you checkout the middle commit, db4c586, you can see the failure by running `rake spec`. I'm more than happy to squash these down, if the maintainers would like.

### CODE REVIEW

**Comment by jaredbeck** [comment]
Thanks Dan. Initial review:

1. Please add a changelog entry under Unreleased -> Fixed
1. I think that the SQL generated by `sibling_versions` should not change, should remain unordered.
1. Unless you feel strongly about keeping it, I don't think we need the "here comes the crazy part" spec. It's a bit hard to follow. If we can replace it with a simpler spec, perhaps by inserting specific values of `created_at`, that'd be fine.

**Comment by danbernier** [comment]
> I think that the SQL generated by sibling_versions should not change, should remain unordered.

So, move the `order` call to the call site, in `enforce_version_limit!`? I can do that.

> ...replace it with a simpler spec, perhaps by inserting specific values of created_at, that'd be fine.

I'll give that a try. Agreed, that'll be much simpler.

**Comment by danbernier** [comment]
@jaredbeck, I made the changes - the CHANGELOG, and moving the `order` out of `sibling_versions`.

I also changed the spec to create fake Version records. It's definitely shorter, but I'm not sure if it's easier to understand. Let me know what you think.

**Comment by jaredbeck** [comment]
Looks good, thanks Dan!

**Comment by jaredbeck** [comment]
Squashed offline into 300a16c

**Comment by danbernier** [comment]
Thank you @jaredbeck! When do you think this fix will reach rubygems?

**Comment by jaredbeck** [comment]
We're very close to the next release, 7.0.0. It could happen this week if all goes well.
